preprocess_web/code/ravens_metadata_apps/preprocess_jobs/job_util.py

```diff
@@ -181,43 +169,68 @@ def retrieve_rows_json(job, **kwargs):
+                        % (num_rows, row_cnt)
+                error_message.append(err)
+                num_rows = row_cnt
+
+        # To read csv given rows count and start rows
         if job.is_tab_source_file():
-            csv_data = pd.read_csv(job.source_file.path,
-                                   sep='\t',
-                                   lineterminator='\r')
-                                   #skiprows=skiprows,
-                                   #nrows=num_rows)
+            try:
+                csv_data = pd.read_csv(job.source_file.path,
+                                       sep='\t',
+                                       lineterminator='\r',
+                                       skiprows=start_row_idx,
+                                       nrows=num_rows)
+
+            except ValueError:
+                print(" not good value for the row start")
+                start_row = 0
+                csv_data = pd.read_csv(job.source_file.path,
+                                       sep='\t',
+                                       lineterminator='\r',
+                                       skiprows=start_row,
+                                       nrows=num_rows)
             print(csv_data)
         elif job.is_csv_source_file():
-            csv_data = pd.read_csv(job.source_file.path)
-                                   #skiprows=skiprows,
-                                   #nrows=num_rows)
+            try:
+                csv_data = pd.read_csv(job.source_file.path,
+                                       skiprows=start_row_idx,
+                                       nrows=num_rows)
+            except ValueError:
+                print(" not good value for the row start")
+                start_row = 0
+                csv_data = pd.read_csv(job.source_file.path,
+                                       skiprows=start_row,
+                                       nrows=num_rows)
+            print(csv_data)
         else:
             return dict(success=False,
                         message='File type unknown (not csv or tab)')
 
-        max_rows = len(csv_data.index)
-        print("the no. of rows are ", max_rows)
-
-        error_message = []
-
-        if start_row > max_rows:
-            err = 'The request was from %s rows but only %d rows were found, so default start rows = 1 is set' % (start_row, max_rows)
-            error_message.append(err)
-            start_row = 1
-        elif num_rows > max_rows:
-            err = 'The request was for %s rows but only %d rows were found, so number rows is set to max rows' % (num_rows, max_rows)
-            error_message.append(err)
-            num_rows = max_rows
-
         update_end_num = start_row + num_rows
         print("error message", error_message)
-        data_frame = csv_data[start_row_idx:update_end_num-1]
+        data_frame = csv_data[start_row_idx:update_end_num - 1]
```

Comment:
Shouldn't need to slice the data; e.g. `csv_data` should be a dataframe with only the rows requested/needed